admin-dev/themes/new-theme/js/product-page/combination.js

```diff
@@ -102,9 +102,14 @@ export default function() {
       $imagesElem.html('');
 
       $.each(combinationsImages[value], function(key, image) {
+        let legend = '';
+        $.each(image.legend, function(k, value) {
+          legend = legend + ' data-legend' + k + '="' + value.replace('"', '') + '"';
```

Comment:
Why not embed all the legends in json format as a single data-attribute? I think it would simplify usage

```html
<img data-legend="{'1':'This is a legend','2':'Another legend'}">
```

```js
let legends = JSON.parse(
  $(this).data('legend')
);
let legend = legends[idLang];

$(this).attr('alt', legend);
$(this).attr('title', legend);
```